deno/src/userWebSockets.ts

```diff
@@ -19,7 +19,9 @@ export function addWebSocket(socket: WebSocket, user?: User) {
   if (currentSocket !== socket) {
     const send = <T extends WebSocketResponse>(data: T) => {
       try {
-        socket.send(JSON.stringify(data))
+        if(socket.readyState === socket.OPEN){
```

Comment:
@mfbx9da4 i've finally come around to take a look at the server code: what was happening was that it tried to send a message before broadcasting, but given that the socket was not open it would've failed to send and call the destroy method that would've also close the channel.

I've added a check for the readystate of the socket before trying to send and this should fix the error that would cause the server to crash.

I was able to find a reproduction if you want to try on your own:

- open a tab and start a race
- open another tab and start a race
- wait for the race to start
- write at least a letter of the race
- close one tab
- refresh the other tab
- start the race again

This is what caused the server to crash before (it will not crash with this fix)

Reply:
> tried to send a message before broadcasting

Isn't the issue that the user left the game and so the web socket connection is now closed and we don't gracefully handle connections being closed?

If the user's connection has dropped, it's fine to just not send messages. So I think your change makes a lot of sense.

Reply:
> Isn't the issue that the user left the game and so the web socket connection is now closed and we don't gracefully handle connections being closed?

Yeah basically the websocket it's closed, in the broadcast method it calls the send method which tries to send a message to the websocket, failing to do so it closes the channel. After the send method the broadcast method tries to post a message but the channel was just closed by the websocket and so it fails.

Reply:
@mfbx9da4 any news? Do i need to do something else or can we merge this?

Reply:
Yeah just one final thing, could you run prettier on your changes? Also there should be a

```
// A space after comments ✅
//Not like this ❌
```

Thank you!

Reply:
All right...did it and pushed a new commit ;)

Reply:
It doesn't look like you ran [prettier](https://prettier.io/)?

Reply:
Uh i don't know what happened...redid it

Reply:
@mfbx9da4 don't know if you missed this